letta/services/message_manager.py

```diff
@@ -0,0 +1,268 @@
+
+        # Can't use enforce_types here, so manually check
+        assert all(isinstance(msg, PydanticMessage) for msg in messages)
+
+        with self.session_maker() as session:
+            msg_models = [MessageModel(**msg.model_dump()) for msg in messages]
+            for msg in msg_models:
+                msg.create(session)
+            return [msg.to_pydantic() for msg in msg_models]
+
+    @enforce_types
+    def update_message_by_id(self, message_id: str, message: PydanticMessage) -> Optional[PydanticMessage]:
+        """
+        Updates an existing record in the database with values from the provided record object.
+
+        Replaces db.py:SQLLiteStorageConnector.update() 
+        """
+        if not message_id:
+            raise ValueError("Message ID must be provided.")
+
+        with self.session_maker() as session:
+            try:
+                # Fetch existing message from database
+                msg = MessageModel.read(db_session=session, identifier=message_id)
+                if not msg:
+                    raise ValueError(f"Message with id {message_id} does not exist.")
+
+                # Update the database record with values from the provided record
+                for column in MessageModel.__table__.columns:
+                    column_name = column.name
+                    if hasattr(message, column_name):
+                        new_value = getattr(message, column_name)
+                        setattr(msg, column_name, new_value)
+
+                # Commit changes
+                msg.update(session)
+                return msg.to_pydantic()
+            except NoResultFound:
+                return None
+
+    @enforce_types
+    def delete_message_by_id(self, message_id: str) -> bool:
+        """Delete a message."""
+        with self.session_maker() as session:
+            try:
+                msg = MessageModel.read(
+                    db_session=session, 
+                    identifier=message_id
+                )
+                msg.hard_delete(session)
+                return True
+            except NoResultFound:
+                return False
+
+    @enforce_types
+    def list_messages(self, user_id: str, agent_id: Optional[str] = None,
+                     cursor: Optional[str] = None, limit: Optional[int] = 50) -> List[PydanticMessage]:
+        """List messages with pagination based on cursor and limit."""
+        with self.session_maker() as session:
+            filters = {"user_id": user_id}
+            if agent_id:
+                filters["agent_id"] = agent_id
+            results = MessageModel.list(
+                        db_session=session, 
+                        cursor=cursor,
+                        limit=limit,
+                        **filters
+                      )
+            return [msg.to_pydantic() for msg in results]
+
+    @enforce_types
+    def query_date_range(self, user_id: str, start_date: datetime, end_date: datetime,
```

Comment:
Could it be possible to combine the following all into `list_messages` (so they all call `MessageModel.list(...)`: 
* `query_text` 
* `get_all` (I think this is redundant to `list_messages`) 

I think for the date range query, we might want to add additional parameters to `SqlalchemyBase.list(..)` to also allow date filters

Reply:
ack